chainer/testing/function_link.py

```diff
@@ -0,0 +1,1116 @@
+        tested. E.g. ``['gamma', 'beta']`` for the batch normalization link.
+        ``[]`` by default.
+
+    ``contiguous`` (None or 'C'):
+        Specifies the contiguousness of incoming arrays (i.e. inputs,
+        parameters and gradients. If ``None``, the
+        arrays will be non-contiguous as long as possible. If ``'C'``, the
+        arrays will be C-contiguous. ``None`` by default.
+
+    .. note::
+
+        This class assumes :func:`chainer.testing.inject_backend_tests`
+        is used together. See the example below.
+
+    .. note::
+
+        When implementing :class:`~chainer.testing.LinkTestCase` and
+        :class:`~chainer.testing.LinkInitializersTestCase` to test both
+        forward/backward and initializers, it is often convenient to refactor
+        out common logic in a separate class.
+
+    .. admonition:: Example
+
+        .. testcode::
+
+            @chainer.testing.inject_backend_tests(
+              None,
+              [
+                  {},  # CPU
+                  {'use_cuda': True},  # GPU
+              ])
+            class TestLinear(chainer.testing.LinkInitializersTestCase):
+
+                param_names = ['W', 'b']
+
+                def generate_params(self):
+                    initialW = numpy.random.uniform(
+                        -1, 1, (3, 2)).astype(numpy.float32)
+                    initial_bias = numpy.random.uniform(
+                        -1, 1, (3,)).astype(numpy.float32)
+                    return initialW, initial_bias
+
+                def generate_inputs(self):
+                    x = numpy.random.uniform(
+                        -1, 1, (1, 2)).astype(numpy.float32)
+                    return x,
+
+                def create_link(self, initializers):
+                    initialW, initial_bias = initializers
+                    link = chainer.links.Linear(
+                        2, 3, initialW=initialW, initial_bias=initial_bias)
+                    return link
+
+                def forward(self, link, inputs, device):
+                    x, = inputs
+                    return link(x),
+
+                def get_initializers(self):
+                    initialW = [initializers.Constant(1), 2]
+                    initial_bias = [initializers.Constant(2), 3,
+                        chainer.testing.link.InitializerArgument(None, 0)]
+                    return initialW, initial_bias
+
+    .. seealso::
+        :class:`~chainer.testing.LinkTestCase`
+        :class:`~chainer.testing.FunctionTestCase`
+
+    """
+
+    check_initializers_options = {}
+
+    def get_initializers(self):
+        raise NotImplementedError('get_initializers is not implemented.')
+
+    def test_initializers(self, backend_config):
+        """Tests that the parameters of a links are correctly initialized."""
+
+        self.before_test('test_initializers')
+
+        params_inits = self._get_initializers()
+
+        for i_param, param_inits in enumerate(params_inits):
```

Comment:
How about leaving TODO to optimize this test? (to minimize the number of iterations)

Reply:
Fixed.